### PR TITLE
fix: use getConfig not process.env

### PR DESCRIPTION
**Pull request by cintnguyen**
Changed process.env to getConfig() in order to support runtime configurations. Moved all process.env use in the index.html header to helmet.

Co-authored-by: @httpsmenahassan

### CODE REVIEW

**Review by deborahgu — CHANGES_REQUESTED** [review]
looks good, but I have a few questions and a request for tests @cintnguyen

File: src/App.jsx [hunk 1]
```diff
@@ -76,6 +90,8 @@ export const App = () => {
     <>
       <Helmet>
         <title>{formatMessage(messages.pageTitle)}</title>
+        <link rel="shortcut icon" href={getConfig().FAVICON_URL} type="image/x-icon" />
```
[the remark this answers is not in the document]
Comment (cintnguyen) [reply src/App.jsx:93]: > was this URL already available from `getConfig`? 

Yes. Based on the [documentation within the `getConfig` file](https://github.com/openedx/frontend-platform/blob/ee6e79b978e99a04f1eb8b8da3ada4eb453813af/src/config.js#L5-L11) 

>  Configuration variables can be supplied to the application in four different ways.  They are applied in the following order:
> - Build-time Configuration
> - Environment Variables
> - JavaScript File
> - Runtime Configuration

The fact that this allows for environment variable base configuration aligns with [how `FAVICON_URL` is being set in `htmlWebpackPlugin.options`](https://github.com/openedx/frontend-build/blob/cf5c12ed421288c519c1486b07028625da197ef1/config/webpack.prod.config.js#L205).

File: src/App.jsx [hunk 2]
```diff
@@ -41,8 +42,21 @@ export const App = () => {
   const { supportEmail } = reduxHooks.usePlatformSettingsData();
   const loadData = reduxHooks.useLoadData();
 
+  const optimizelyScript = () => {
+    if (getConfig().OPTIMIZELY_URL) {
+      return <script src={getConfig().OPTIMIZELY_URL} />;
```
[the remark this answers is not in the document]
Comment (cintnguyen) [reply src/App.jsx:47]: Tried tackling this in https://github.com/openedx/frontend-app-learner-dashboard/commit/370aa3f5c32a54ad0be2a89c46d145c72f1490b7, ran into issues with `runBasicTests()` not respecting dynamic `getConfig` mocks. It seems to be a test scope issue. I am not quite sure what refactoring of App.test.jsx would be needed to get this working yet.

I will be out of office until end of next week, in the meantime @brian-smith-tcril and @httpsmenahassan have offered to take a look at this.

File: public/index.html [hunk 3]
```diff
@@ -1,19 +1,8 @@
 <!doctype html>
 <html lang="en-us" dir="ltr">
   <head>
-      <title>Learner Dashboard | <%= process.env.SITE_NAME %></title>
```
[the remark this answers is not in the document]
Comment (arbrandes) [reply public/index.html:4]: I'm not aware of a specific reason, except for the fact it comes from the pre-`Helmet` days.  I would suggest simply removing it.  We can revisit this easily if it becomes an issue.

**Comment by brian-smith-tcril** [comment]
@deborahgu I added tests that cover the lines you mentioned, but it seems the overall coverage is still down from before.

The only thing that stood out to me when looking at the diff between `App.jsx` coverage on `master` vs this PR was hotjar

### `master`
image

### PR
image

but I'm not sure what's testing this `useEffect()` block.

I'll gladly update whatever test was covering these lines before, I just don't know where I should be looking.

**Review by deborahgu — APPROVED** [review]
> I'll gladly update whatever test was covering these lines before, I just don't know where I should be looking.

yeah, @brian-smith-tcril, I'm a little stumped by this as well. I'd think it's just the `getConfig` on line 77 it wants tested, but then it's complaining about 2 lines, not one.

I'm comfortable forcing this through with the codecov complaint, tbh. I won't merge until morning because I hate doing something weird when it's the end of the day, but I'm going to approve this.

**Comment by brian-smith-tcril** [comment]
> I'd think it's just the getConfig on line 77 it wants tested, but then it's complaining about 2 lines, not one.

That part makes sense to me, it's testing the condition but it's never testing it *with a* `HOTJAR_APP_ID`. Before the switch to `getConfig` it would pull that from here

https://github.com/openedx/frontend-app-learner-dashboard/blob/5ca1e9dc1f7cbcbd7de6519229e0fdd0e8994399/.env.test#L42-L44

I tried adding hotjar stuff to the mocked return values for `getConfig` in the same way that I handled `OPTIMIZELY_URL` but codecov still complained about those lines. That makes me think that `App.test.jsx` doesn't include any tests that trigger [that `useEffect`](https://github.com/openedx/frontend-app-learner-dashboard/blob/5ca1e9dc1f7cbcbd7de6519229e0fdd0e8994399/src/App.jsx#L44-L74)

```jsx
  React.useEffect(() => {
    // [...]
  }, [authenticatedUser, loadData]);
```

Since `HOTJAR_APP_ID` was being set in *every* test before (by means of existing in `.env.test`) that means the test that *was* covering those lines could be *anything*.

I was hoping I could find a way to see what test covered what lines, but I didn't see anything the codecov web UI or the html report generated by running jest locally. My searches ("jest coverage see what test is covering a line" and "codecov see what test is covering a line") also didn't lead anywhere helpful.

Hopefully I can figure that out at some point. It seems like a very useful thing when updating tests in scenarios like this.

___

> I'm comfortable forcing this through with the codecov complaint, tbh. I won't merge until morning because I hate doing something weird when it's the end of the day, but I'm going to approve this.

### Thank you so much!

**Pull request merged**